### DISCUSSION TITLE
Usuwanie shipping-rates

### DISCUSSION THREAD

**Discussion by fejmek** [allegro/allegro-api#9415] [Suggestions - rejected]
:bulb: **Sugestia / Suggestions**

Proponuje dodanie endpointa do usuwania shipping-rates.

Dlaczego?  
W celu dostosowania do każdego produktu odpowiedniej metody wysyłki, generujemy wszystkie możliwe sposoby wysyłki dla konkretnych rozmiarów paczek. W momencie jak zrobimy na ten moment 40 metod (w przyszłości ok. 700), najmniejszy błąd czy potrzebna zmiana lub usunięcie już nie istniejącego wariantu jest dość trudne, bo robi się to ręcznie w panelu i pozbawione sensu, gdy mamy API.

**Comment by MartaNowaczyk** [comment]
Sugestie zgłaszamy w zakładce "Discussion", więc pozwoliłam sobie przenieść twoją tutaj. Przekazałam ją do odpowiedniego zespołu, także na ten moment zmieniłam status na "Suggestion - verification".

**Reply by fejmek** [reply]
Pardon i dziękuję za przeniesienie.
Czy w ogóle dobrze myślę w tej kwestii? jak wygląda sprawa przypisywania różnych metod dostaw do różnych rozmiarów produktów, omijam jakas funkcjonalność, że muszę tak kombinować z cennikami?

**Reply by MartaNowaczyk** [reply]
Ogólnie, [cenniki dostaw](https://developer.allegro.pl/tutorials/jak-zarzadzac-kontem-danymi-uzytkownika-ZM9YAKgPgi2#jak-pobrac-cennik-dostaw) można edytować przez nasze API. Nie można ich tylko usuwać przez API. 
To prawda, różne przedmioty mogę wymagać różnych metod dostawy. Pamiętaj też, że możesz dodać maksymalnie 250 cenników.  Ogólne zasady znajdziesz na [dedykowanej stronie](https://help.allegro.com/sell/pl/a/cenniki-dostawy-tworzenie-edycja-i-podmiana-1nPyDZBbPtd).

**Comment by MartaNowaczyk** [comment]
Mam informację, że obecnie nie planujemy udostępnienia opcji usuwania cenników z poziomu API, stąd sugestię musimy odrzucić.

**Comment by fejmek** [comment]
Dziękuję za odpowiedź, ale nie będę ukrywał, że bardzo źle to brzmi. 

Dostajemy ponad 15 metod dostawy, wraz z ich opcjami, głównie patrz Maksymalna ilość produktów w paczce i opłata, za każda następną sztukę. 
Godzę się z faktem, że nie możemy ustawić w produkcie każdej metody oddzielne, co mniej więcej mogę jedynie sobie wyobrazić, jaki byłby problem i zamieszanie, żeby to teraz zmienić. Lecz przynajmniej nie dawajcie nam limitów na ilość cenników. 
Wyobrażam sobie, że pewnie nie jedna mądra głowa wpadła na podział cenników w ten sposób.

Ustawiając wiele metod dostaw, nie da się zmieścić w 250 sztukach cenników, uwzględniając w każdej metodzie cene i ilosc max sztuk w paczce, za które klient nie płaci. Rozmiary przesyłek są różne tak samo jak ich wagi, każdy kurier wymaga co innego.

Sklep traci przede wszystkim pieniądze i zadowolonych klientów, którzy mogą mieć problem z zakupem wielosztuk, ponieważ za każda zapłaci dodatkowe cene wysyłki.
Klient, który płaci za wysyłkę tyle ile powinien, to zadowolony klient. :)

Wnioskuję chociaż od zwiększenie limitów cenników.

**Reply by MartaNowaczyk** [reply]
Rozumiem, przekażę do zespołu sugestię dotyczącą zmiany limitu cenników.
